platform/mbed_retarget.cpp

```diff
@@ -120,6 +121,17 @@ static void init_serial() {
 #endif
 }
 
+/**
+ * Sets errno when file opening fails.
+ * Wipes out the filehandle too.
+ */
+static int handle_open_errors(int error, unsigned filehandle_idx) {
+    errno = -error;
```

Comment:
This is a bit bigger than this pr. Error reporting (using negative errno codes) should probably get its own page in the handbook.